db/redis/db.go

```diff
@@ -0,0 +1,496 @@
+	data, err := json.Marshal(values)
+	if err != nil {
+		return err
+	}
+
+	switch r.mode {
+	case "cluster":
+		err = r.clusterCli.Set(table+"/"+key, string(data), 0).Err()
+	case "single":
+		fallthrough
+	default:
+		err = r.singleCli.Set(table+"/"+key, string(data), 0).Err()
+	}
+
+	return err
+}
+
+func (r *redis) Delete(ctx context.Context, table string, key string) error {
+	var err error
+
+	switch r.mode {
+	case "cluster":
+		err = r.clusterCli.Del(table + "/" + key).Err()
+	case "single":
+		fallthrough
+	default:
+		err = r.singleCli.Del(table + "/" + key).Err()
+	}
+
+	return err
+}
+
+type redisCreator struct{}
+
+func (r redisCreator) Create(p *properties.Properties) (ycsb.DB, error) {
+	rds := &redis{}
+
+	mode, _ := p.Get(redisMode)
+	switch mode {
+	case "cluster":
+		rds.clusterCli = goredis.NewClusterClient(getOptionsCluster(p))
+
+		if p.GetBool(prop.DropData, prop.DropDataDefault) {
+			err := rds.clusterCli.FlushDB().Err()
+			if err != nil {
+				return nil, err
+			}
+		}
+	case "single":
+		fallthrough
+	default:
+		mode = "single"
+		rds.singleCli = goredis.NewClient(getOptionsSingle(p))
+
+		if p.GetBool(prop.DropData, prop.DropDataDefault) {
+			err := rds.singleCli.FlushDB().Err()
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	rds.mode = mode
+
+	return rds, nil
+}
+
+const (
+	redisMode                  = "redis.mode"
+	redisNetwork               = "redis.network"
+	redisAddr                  = "redis.addr"
+	redisPassword              = "redis.password"
+	redisDB                    = "redis.db"
+	redisMaxRedirects          = "redis.max_redirects"
+	redisReadOnly              = "redis.read_only"
+	redisRouteByLatency        = "redis.route_by_latency"
+	redisRouteRandomly         = "redis.route_randomly"
+	redisMaxRetries            = "redis.max_retries"
+	redisMinRetryBackoff       = "redis.min_retry_backoff"
+	redisMaxRetryBackoff       = "redis.max_retry_backoff"
+	redisDialTimeout           = "redis.dial_timeout"
+	redisReadTimeout           = "redis.read_timeout"
+	redisWriteTimeout          = "redis.write_timeout"
+	redisPoolSize              = "redis.pool_size"
+	redisMinIdleConns          = "redis.min_idle_conns"
+	redisMaxConnAge            = "redis.max_conn_age"
+	redisPoolTimeout           = "redis.pool_timeout"
+	redisIdleTimeout           = "redis.idle_timeout"
+	redisIdleCheckFreq         = "redis.idle_check_frequency"
+	redisTLSCA                 = "redis.tls_ca"
+	redisTLSCert               = "redis.tls_cert"
+	redisTLSKey                = "redis.tls_key"
+	redisTLSInsecureSkipVerify = "redis.tls_insecure_skip_verify"
+)
+
+func getOptionsSingle(p *properties.Properties) *goredis.Options {
+	opts := &goredis.Options{}
+	opts.Network, _ = p.Get(redisNetwork)
+	opts.Addr, _ = p.Get(redisAddr)
+	opts.Password, _ = p.Get(redisPassword)
+	opts.DB = p.GetInt(redisDB, 0)
+	opts.MaxRetries = p.GetInt(redisMaxRetries, 0)
+
+	var err error
+	tmp, ok := p.Get(redisMinRetryBackoff)
```

Comment:
why not using https://godoc.org/github.com/magiconair/properties#Properties.GetDuration directly?

Reply:
Huh, you are right there are `GetDuration` and `GetParsedDuration`, must've been missed it.